README.md

```diff
@@ -82,32 +93,71 @@ func (person *Person) GrowOlder() {
 }
 ```
 
-Internally the `TrackChange` functions calls the `Transition` function on the aggregate to transform the aggregate based on the newly created event.
+Internally the `TrackChange` methods calls the `Transition` method on the aggregate to transform the aggregate based on the newly created event.
 
-To bind metadata to events use the `TrackChangeWithMetadata` function.
+To bind metadata to events use the `TrackChangeWithMetadata` method.
   
-
 The internal `Event` looks like this.
 
 ```go
 type Event struct {
     // aggregate identifier 
-    AggregateID string
+    aggregateID string
     // the aggregate version when this event was created
-    Version         Version
+    version         Version
     // the global version is based on all events (this value is only set after the event is saved to the event store) 
-    GlobalVersion   Version
+    globalVersion   Version
     // aggregate type (Person in the example above)
-    AggregateType   string
+    aggregateType   string
     // UTC time when the event was created  
-    Timestamp       time.Time
+    timestamp       time.Time
     // the specific event data specified in the application (Born{}, AgedOneYear{})
-    Data            interface{}
+    data            interface{}
     // data that don´t belongs to the application state (could be correlation id or other request references)
-    Metadata        map[string]interface{}
+    metadata        map[string]interface{}
 }
 ```
 
+To access properties on the event there are methods exposing them. This prevent external parties to modify the event from the outside.
+
+```go
+    func (e Event) AggregateID() string {
+        return e.event.AggregateID
+    }
+
+    func (e Event) Version() Version {
+        return Version(e.event.Version)
+    }
+
+    func (e Event) GlobalVersion() Version {
+        return Version(e.event.GlobalVersion)
+    }
+
+    func (e Event) AggregateType() string {
+        return e.event.AggregateType
+    }
+
+    func (e Event) Timestamp() time.Time {
+        return e.event.Timestamp
+    }
+
+    func (e Event) Data() interface{} {
+        return e.data
+    }
+
+    func (e Event) Metadata() map[string]interface{} {
+        return e.metadata
+    }
+
+    // Reason is based on the data type
+    func (e Event) Reason() string {
+        if e.data == nil {
+            return ""
+        }
+        return reflect.TypeOf(e.data).Elem().Name()
+    }
+```
+
```

Comment:
I think it's preferable if you could simply mention that there exists methods for accessing these fields, instead of having a copy of the actual code in the documentation. If you change the code, you should also update this.

```suggestion
To access properties on the event you can use the corresponding methods exposing them, e.g `AggregateID()`. This prevent external parties to modify the event from the outside.
```